Update based on legal requirements. (#900)
huggingface/optimum-habana: README.md

![](https://github.com/huggingface/optimum-habana/blob/main/readme_logo.png)

> # Optimum for Intel® Gaudi® Accelerator 

Reviewer: ```suggestion
# Optimum for Intel® Gaudi® Accelerators 
```

> # Optimum for Intel(R) Gaudi(R) Accelerator 

Author: @regisss  do you think the R can be changed to trade mark version?

Reviewer 2: Can you try `:registered:`? This should give :registered:
Or are you talking about :tm:?

Author: do I need ( ) around it?

Reviewer 2: I don't think so

Author: seems not working, I have to copy it somehow

> 🤗 Optimum for Intel® Gaudi® accelerator is the interface between the 🤗 Transformers and Diffusers libraries and [Intel Gaudi AI accelerator (HPU)](https://docs.habana.ai/en/latest/index.html).

Reviewer: ```suggestion
Optimum for Intel Gaudi - a.k.a. `optimum-habana` - is the interface between the Transformers and Diffusers libraries and [Intel® Gaudi® AI Accelerators (HPU)](https://docs.habana.ai/en/latest/index.html).
```

> Optimum for Intel Gaudi - a.k.a. `optimum-habana` - is the interface between the Transformers and Diffusers libraries and [Intel® Gaudi® AI Accelerators (HPU)](https://docs.habana.ai/en/latest/index.html).

Reviewer 2: ```suggestion
Optimum for Intel Gaudi - a.k.a. `optimum-habana` - is the interface between the Transformers and Diffusers libraries and [Intel Gaudi AI Accelerators (HPU)](https://docs.habana.ai/en/latest/index.html).
```
The registration marks already appear in the title.

It provides a set of tools enabling easy model loading, training and inference on single- and multi-HPU settings for different downstream tasks.

> The list of officially validated models and tasks is available [here](https://github.com/huggingface/optimum-habana#validated-models). Users can try other of the thousands of Hugging Face models on Intel® Gaudi accelerators and tasks with only few changes.

Reviewer 2: ```suggestion
The list of officially validated models and tasks is available [here](https://github.com/huggingface/optimum-habana#validated-models). Users can try other of the thousands of Hugging Face models on Intel Gaudi accelerators and tasks with only few changes.
```

> ## What is a Intel® Gaudi® AI accelerator (HPU)?

Reviewer: ```suggestion
## What are Intel Gaudi AI Accelerators (HPUs)?
```

HPUs offer fast model training and inference as well as a great price-performance ratio.

> Check out [this blog post about BLOOM inference](https://huggingface.co/blog/habana-gaudi-2-bloom) and [this article benchmarking Intel® Gaudi® accelerator Gaudi2 versus Nvidia A100 GPUs for BridgeTower training](https://huggingface.co/blog/bridgetower) for concrete examples.

Reviewer: ```suggestion
Check out [this blog post about BLOOM inference](https://huggingface.co/blog/habana-gaudi-2-bloom) and [this post benchmarking Intel Gaudi 2 and NVIDIA A100 GPUs for BridgeTower training](https://huggingface.co/blog/bridgetower) for concrete examples.
```

### Option 2: Use the latest main branch under development

> Optimum for Intel® Gaudi® accelerator is a fast-moving project, and you may want to install it from source and get the latest scripts :

Reviewer: ```suggestion
Optimum for Intel Gaudi is a fast-moving project, and you may want to install it from source and get the latest scripts :
```

### Option 2: Use the latest main branch under development

> Optimum Intel(R) Gaudi(R) accelerator is a fast-moving project, and you may want to install it from source and get the latest scripts :

Reviewer 2: Here I would prefer to keep the same formulation as the title of the README. It is supposed to mention the name of the library and that's fine if we want to remove the "Habana" mention but I feel like "Optimum Intel(R) Gaudi(R) accelerator" doesn't make much sense as it has never been advertised that way. So I would rather have "Optimum for Intel(R) Gaudi(R) accelerator".
```suggestion
Optimum for Intel(R) Gaudi(R) accelerator is a fast-moving project, and you may want to install it from source and get the latest scripts :
```

### Quick Start

> 🤗 Optimum for Intel® Gaudi® accelerator was designed with one goal in mind: **to make training and inference straightforward for any 🤗 Transformers and 🤗 Diffusers user while leveraging the complete power of the Intel® Gaudi® AI accelerator**.

Reviewer: ```suggestion
Optimum for Intel Gaudi was designed with one goal in mind: **to make training and inference straightforward for Transformers and Diffusers users, while fully leveraging the power of Intel Gaudi AI Accelerators**.
```

#### Transformers Interface

There are two main classes one needs to know:
- [GaudiTrainer](https://huggingface.co/docs/optimum/habana/package_reference/trainer): the trainer class that takes care of compiling and distributing the model to run on HPUs, and performing training and evaluation.
- [GaudiConfig](https://huggingface.co/docs/optimum/habana/package_reference/gaudi_config): the class that enables to configure Habana Mixed Precision and to decide whether optimized operators and optimizers should be used or not.

> The [GaudiTrainer](https://huggingface.co/docs/optimum/habana/package_reference/trainer) is very similar to the [🤗 Transformers Trainer](https://huggingface.co/docs/transformers/main_classes/trainer), and adapting a script using the Trainer to make it work with Intel® Gaudi® Accelerator will mostly consist in simply swapping the `Trainer` class for the `GaudiTrainer` one.

Reviewer: ```suggestion
The [GaudiTrainer](https://huggingface.co/docs/optimum/habana/package_reference/trainer) is very similar to the [Transformers Trainer](https://huggingface.co/docs/transformers/main_classes/trainer), and adapting a script using the Trainer to make it work with Intel Gaudi accelerators will mostly consist in simply swapping the `Trainer` class for the `GaudiTrainer` one.
```

### Quick Start

> 🤗 Optimum Intel(R) Gaudi(R) accelerator was designed with one goal in mind: **to make training and inference straightforward for any 🤗 Transformers and 🤗 Diffusers user while leveraging the complete power of the Intel(R) Gaudi(R) AI accelerator**.

Reviewer 2: ```suggestion
🤗 Optimum for Intel(R) Gaudi(R) accelerator was designed with one goal in mind: **to make training and inference straightforward for any 🤗 Transformers and 🤗 Diffusers user while leveraging the complete power of the Intel(R) Gaudi(R) AI accelerator**.
```
Same

)
```

> where `gaudi_config_name` is the name of a model from the [Hub](https://huggingface.co/Habana) (Intel® Gaudi® accelerator configurations are stored in model repositories) or a path to a local Intel® Gaudi® accelerator configuration file (you can see [here](https://huggingface.co/docs/optimum/habana/package_reference/gaudi_config) how to write your own).

Reviewer: ```suggestion
where `gaudi_config_name` is the name of a model from the [Hub](https://huggingface.co/Habana) (Intel Gaudi configurations are stored in model repositories) or a path to a local Intel Gaudi configuration file (you can see [here](https://huggingface.co/docs/optimum/habana/package_reference/gaudi_config) how to write your own).
```

#### Diffusers Interface

> You can generate images from prompts using Stable Diffusion on Intel® Gaudi® accelerator using the [`GaudiStableDiffusionPipeline`](https://huggingface.co/docs/optimum/habana/package_reference/stable_diffusion_pipeline) class and the [`GaudiDDIMScheduler`] which have been both optimized for HPUs. Here is how to use them and the differences with the 🤗 Diffusers library:

Reviewer: ```suggestion
You can generate images from prompts using Stable Diffusion on Intel Gaudi using the [`GaudiStableDiffusionPipeline`](https://huggingface.co/docs/optimum/habana/package_reference/stable_diffusion_pipeline) class and the [`GaudiDDIMScheduler`] which have been both optimized for HPUs. Here is how to use them and the differences with the Diffusers library:
```

### Documentation

> Check out [the documentation of Optimum for Intel® Gaudi® accelerator](https://huggingface.co/docs/optimum/habana/index) for more advanced usage.

Reviewer: ```suggestion
Check out [the documentation of Optimum for Intel Gaudi](https://huggingface.co/docs/optimum/habana/index) for more advanced usage.
```

## Validated Models

> The following model architectures, tasks and device distributions have been validated for 🤗 Optimum for Intel® Gaudi® accelerator:

Reviewer: ```suggestion
The following model architectures, tasks and device distributions have been validated for Optimum for Intel Gaudi:
```

</div>

> Other models and tasks supported by the 🤗 Transformers and 🤗 Diffusers library may also work. You can refer to this [section](https://github.com/huggingface/optimum-habana#how-to-use-it) for using them with 🤗 Optimum for Intel® Gaudi® accelerator. Besides, [this page](https://github.com/huggingface/optimum-habana/tree/main/examples) explains how to modify any [example](https://github.com/huggingface/transformers/tree/main/examples/pytorch) from the 🤗 Transformers library to make it work with 🤗 Optimum for Intel® Gaudi® accelerator.

Reviewer: ```suggestion
Other models and tasks supported by the Transformers and Diffusers libraries may also work. You can refer to this [section](https://github.com/huggingface/optimum-habana#how-to-use-it) for using them with Optimum for Intel Gaudi. In addition, [this page](https://github.com/huggingface/optimum-habana/tree/main/examples) explains how to modify any [example](https://github.com/huggingface/transformers/tree/main/examples/pytorch) from the Transformers library to make it work with Optimum for Intel Gaudi.
```

If you find any issues while using those, please open an issue or a pull request.

## Gaudi Setup

> Please refer to the Intel® Gaudi® accelerator's official [installation guide](https://docs.habana.ai/en/latest/Installation_Guide/index.html).

Reviewer: ```suggestion
Please refer to the Intel® Gaudi® AI Accelerator official [installation guide](https://docs.habana.ai/en/latest/Installation_Guide/index.html).
```

Reviewer 2: ```suggestion
Please refer to the Intel Gaudi AI Accelerator official [installation guide](https://docs.habana.ai/en/latest/Installation_Guide/index.html).
```

> > Tests should be run in a Docker container based on Intel® Gaudi® accelerator Docker images.

Reviewer: ```suggestion
> Tests should be run in a Docker container based on Intel Gaudi Docker images.
```
